### PR TITLE
Fix: Edge to edge

### PR DESCRIPTION
**Pull request by KCeh**
## Summary

Fix for issue https://github.com/infinum/android-sentinel/issues/86
Now Sentinel's UI should be ok on API 35 devices in regards to edge to edge

## Changes

* Bumped compile and target SDK to 35
* Fixed compile issues with SDK 35
* Bumped Chucker to version that has fixed edge to edge
* Fixed edge to edge in Sentinel fullscreen screens `android:fitsSystemWindows="true"`

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [x] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

As mentioned I focused on most intrusive part of issue (you can't see/click on top bar).
There might be room to improve things even further: sample app edge to edge support? bottom padding support? and bottom sheet is now floating a bit weirdly.
If you have suggestions how to improve things, feel free to mention them 

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

## Additional notes

**Before:** 
before-6
before-5
before-4
before-3
before-2
before-1

**After:**
[after.webm](https://github.com/user-attachments/assets/f6cafef1-6149-47c7-9a17-557a3db8d21e)

### CODE REVIEW

**Review by AsimRibo — APPROVED** [review]
✅ 
Activity methods signatures were changed so I had to update them on the other project.
This solves a good portion of edge to edge problems but we will need to add additional updates later.

File: sentinel/src/main/res/layout/sentinel_fragment_settings.xml [hunk 1]
```diff
@@ -8,6 +8,7 @@
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:background="?android:colorBackground"
+        android:fitsSystemWindows="true"
```
Comment (AsimRibo) [inline sentinel/src/main/res/layout/sentinel_fragment_settings.xml:11]: Shouldn't fitsSystemWindows go to root parent so it handles both top and bottom insets?

**Comment by AsimRibo** [comment]
> > This solves a good portion of edge to edge problems but we will need to add additional updates later.
> 
> We are in no rush, so maybe we can handle everything here? What needs to be done?
> 
> > Shouldn't fitsSystemWindows go to root parent so it handles both top and bottom insets?
> 
> 🤔 we can test it that way

In Sentinel `BaseActivity` we need to call `enableEdgeToEdge()`, otherwise edge to edge will be enforced only on Android 15 by default. Currently, Android < 15 is still not edge to edge.

I double checked regarding `fitsSystemWindows` and from what I see, it should be in root parent view but that is causing us troubles with system toolbar so I wouldn't touch it right now.  As you can see in video below, when edge to edge is enabled system navigation bar appears semi-transparent but it's not a huge deal.

We also need to figure out why toolbar is changing color, I don't think this was intended, but looks ok for now.

Take_Screenshot


Same steps can be applied to sample app as well.

**Comment by KCeh** [comment]
> In Sentinel `BaseActivity` we need to call `enableEdgeToEdge()`, otherwise edge to edge will be enforced only on Android 15 by default.

I mean we are not the app but the lib. We only want to display properly UI for apps that support edge-to-edge.
Meaning: 
I have Android 14 without edge-to-edge -> Sentinel is not edge-to-edge, UI looks ok -> ✅ 
I have Android 15 with edge-to-edge -> Sentinel is edge-to-edge, UI look ok -> ✅
